src/pivot-table/components/cells/ColumnAdjuster.tsx

```diff
@@ -0,0 +1,76 @@
+import { preventDefaultBehavior } from "@qlik/nebula-table-utils/lib/utils";
+import React, { useEffect, useRef, useState } from "react";
+import { ColumnWidthType } from "../../../types/QIX";
+import type { Cell, DataModel } from "../../../types/types";
+import { GRID_BORDER } from "../../constants";
+import { ColumnWidthValues } from "../../hooks/use-column-width";
+import { CELL_PADDING } from "../shared-styles";
+import { AdjusterBorder, AdjusterHitArea } from "./styles";
+
+interface AdjusterProps {
+  cell: Cell;
+  columnWidth: number;
+  dataModel: DataModel | undefined;
+}
+
+/**
+ * Component that is placed on top of column border, to resize the columns.
+ * When you start dragging, mouse move and mouse up listeners are added.
+ * While dragging this components follows the pointer, and on mouse up all column widths are updated.
+ */
+const ColumnAdjuster = ({ cell, columnWidth, dataModel }: AdjusterProps) => {
+  const [internalWidth, setInternalWidth] = useState(columnWidth);
+  const tempWidth = useRef({ initWidth: 0, columnWidth: 0, initX: 0 });
+  const LEFT_ADJUSTMENT = CELL_PADDING + GRID_BORDER;
+
+  useEffect(() => setInternalWidth(columnWidth), [columnWidth]);
+
+  const mouseMoveHandler = (evt: MouseEvent) => {
+    const deltaWidth = evt.clientX - tempWidth.current.initX;
+    const adjustedWidth = Math.max(tempWidth.current.initWidth + deltaWidth, ColumnWidthValues.PixelsMin);
+    setInternalWidth(adjustedWidth);
+    tempWidth.current.columnWidth = adjustedWidth;
+  };
+
+  const mouseUpHandler = (evt: MouseEvent) => {
+    preventDefaultBehavior(evt);
+    document.removeEventListener("mousemove", mouseMoveHandler);
+    document.removeEventListener("mouseup", mouseUpHandler);
+
+    if (tempWidth.current.columnWidth !== tempWidth.current.initWidth) {
+      const newWidthData = { type: ColumnWidthType.Pixels, pixels: tempWidth.current.columnWidth };
+      dataModel?.applyColumnWidth(newWidthData, cell);
+    }
+  };
+
+  const mouseDownHandler = (evt: React.MouseEvent) => {
+    evt.stopPropagation();
+    evt.preventDefault();
+
+    tempWidth.current = {
+      initX: evt.clientX,
+      initWidth: columnWidth,
+      columnWidth,
+    };
+
+    document.addEventListener("mousemove", mouseMoveHandler);
+    document.addEventListener("mouseup", mouseUpHandler);
+  };
+
+  const handleDoubleClick = () => dataModel?.applyColumnWidth({ type: ColumnWidthType.FitToContent }, cell);
```

Comment:
Love this one, feels really nice to use ❤️‍🔥

Reply:
Thank microsoft for inventing it and edvard for implementing it in the old table I guess ;)